docs/src/main/asciidoc/optaplanner.adoc

```diff
@@ -0,0 +1,1034 @@
+            solution = solverJob.getFinalBestSolution();
+        } catch (InterruptedException | ExecutionException e) {
+            throw new IllegalStateException("Solving failed.", e);
+        }
+        return solution;
+    }
+
+}
+----
+
+For simplicity's sake, this initial implementation waits for the solver to finish,
+which can still cause an HTTP timeout.
+The _complete_ implementation avoids HTTP timeouts much more elegantly.
+
+== Set the termination time
+
+Without a termination setting or a termination event, the solver runs forever.
+To avoid that, limit the solving time to five seconds.
+That is short enough to avoid the HTTP timeout.
+
+Create the `src/main/resources/application.properties` file:
+
+[source,properties]
+----
+# The solver runs only for 5 seconds to avoid a HTTP timeout in this simple implementation.
+# It's recommended to run for at least 5 minutes ("5m") otherwise.
+quarkus.optaplanner.solver.termination.spent-limit=5s
+----
+
+
+== Make the application executable
+
+First start the application:
+
+----
+./mvnw compile quarkus:dev
+----
+
+=== Try the application
+
+Now that the application is running, you can test the REST service.
+You can use any REST client you wish.
+The following example uses the Linux command `curl` to send a POST request:
+
+----
+$ curl -i -X POST http://localhost:8080/timeTable/solve -H "Content-Type:application/json" -d '{"timeslotList":[{"dayOfWeek":"MONDAY","startTime":"08:30:00","endTime":"09:30:00"},{"dayOfWeek":"MONDAY","startTime":"09:30:00","endTime":"10:30:00"}],"roomList":[{"name":"Room A"},{"name":"Room B"}],"lessonList":[{"id":1,"subject":"Math","teacher":"A. Turing","studentGroup":"9th grade"},{"id":2,"subject":"Chemistry","teacher":"M. Curie","studentGroup":"9th grade"},{"id":3,"subject":"French","teacher":"M. Curie","studentGroup":"10th grade"},{"id":4,"subject":"History","teacher":"I. Jones","studentGroup":"10th grade"}]}'
+----
+
+After about five seconds, according to the termination spent time defined in your `application.properties`,
+the service returns output similar to the following example:
+
+----
+HTTP/1.1 200
+Content-Type: application/json
+...
+
+{"timeslotList":...,"roomList":...,"lessonList":[{"id":1,"subject":"Math","teacher":"A. Turing","studentGroup":"9th grade","timeslot":{"dayOfWeek":"MONDAY","startTime":"08:30:00","endTime":"09:30:00"},"room":{"name":"Room A"}},{"id":2,"subject":"Chemistry","teacher":"M. Curie","studentGroup":"9th grade","timeslot":{"dayOfWeek":"MONDAY","startTime":"09:30:00","endTime":"10:30:00"},"room":{"name":"Room A"}},{"id":3,"subject":"French","teacher":"M. Curie","studentGroup":"10th grade","timeslot":{"dayOfWeek":"MONDAY","startTime":"08:30:00","endTime":"09:30:00"},"room":{"name":"Room B"}},{"id":4,"subject":"History","teacher":"I. Jones","studentGroup":"10th grade","timeslot":{"dayOfWeek":"MONDAY","startTime":"09:30:00","endTime":"10:30:00"},"room":{"name":"Room B"}}],"score":"0hard/0soft"}
+----
```

Comment:
I wonder if you should use a JSON formatter to format the output and make things more readable? You could add `jq` as a prerequisite and use it to format the JSON outputs?

Reply:
I also thought about it, seeing it in other examples, but I am not sure if it's worth the cost:
- an extra prerequisite (a hoop to jump through to get started)
- that json is already abbreviated, notice the "..." in there - it doesn't help parsing
- a pretty json is still no visualization of the schedule: it's impractical to see broken hard constraints in there. A UI is needed to visualize the schedule (optaplanner != hibernate in that aspect), that's why we have that JS file in the quickstart...

I feel the pain though.